synedrion/src/cggmp21/entities.rs

```diff
@@ -21,18 +21,26 @@ use crate::{
 
 /// The result of the KeyInit protocol.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct KeyShare<P, I: Ord> {
+#[serde(bound(deserialize = "I: for<'x> Deserialize<'x>"))]
+pub struct KeyShare<P, I>
+where
+    P: SchemeParams,
+    I: Ord,
+{
     owner: I,
     /// Secret key share of this node.
-    secret: Secret<Scalar>, // `x_i`
+    secret: Secret<Scalar<P>>, // `x_i`
     public: PublicKeyShares<P, I>, // `X_j`
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct PublicKeyShares<P, I: Ord> {
-    shares: BTreeMap<I, Point>, // `X_j`
-    // TODO (#27): this won't be needed when Scalar/Point are a part of `P`
-    phantom: PhantomData<P>,
+#[serde(bound(deserialize = "I: for<'x> Deserialize<'x>"))]
+pub struct PublicKeyShares<P, I>
+where
+    P: SchemeParams,
+    I: Ord,
+{
+    shares: BTreeMap<I, Point<P>>, // `X_j`
```

Comment:
Maybe at this point this should just be a wrapper over `BTreeMap`, `pub struct PublicKeyShares<P, I>(BTreeMap<I, Point<P>>)`?

Reply:
Yeah, probably.